pkg/scaffold/template.go

```diff
@@ -0,0 +1,76 @@
+package scaffold
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/oslokommune/okctl/pkg/storage"
+
+	kaex "github.com/oslokommune/kaex/pkg/api"
+)
+
+// FetchTemplate downloads an example file and writes it to a buffer
+func FetchTemplate(kx kaex.Kaex) ([]byte, error) {
+	var (
+		err    error
+		buffer bytes.Buffer
+	)
+
+	err = kaex.FetchTemplate(kx, &buffer, "application")
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch example: %w", err)
+	}
+
+	return buffer.Bytes(), nil
+}
+
+// InterpolationOpts defines possible data to inject into the templates
+type InterpolationOpts struct {
+	Domain string
+}
+
+// InterpolateTemplate replaces dummy data in the template with state dependant data
+func InterpolateTemplate(template []byte, opts *InterpolationOpts) ([]byte, error) {
```

Comment:
In engineering and science, the word [interpolate](https://en.wikipedia.org/wiki/Interpolation) is a method of estimation. For instance in the number series "1 2 x 4 5", we can interpolate that x=3, but it could very well actually be something completely different. In this function we are not estimating anything, we are replacing values.

However, I see the [dictionary](https://www.merriam-webster.com/dictionary/interpolate) says "to insert between other things or parts", which doesn't sound too far off, except that it says insert and not replace.

if you follow these thoughts, we should consider some other naming. What do you think?

Reply:
I see the term interpolation used when dealing with templates to inject values where there is variables, thats why I named it that.

Ref: https://medium.com/@channaly/string-interpolation-in-golang-ecb3bcb2d600
https://intellij-support.jetbrains.com/hc/en-us/community/posts/360007837900-How-do-you-toggle-template-interpolation-in-Helm-chart-templates-
https://angular.io/guide/interpolation
https://pugjs.org/language/interpolation.html
etccccc

I felt it hit the nail on the head, but sure - if its a confusing term I can change it. Want me to change it? @yngvark @ivaruf could chime in as well

Reply:
Okay, it seems string interpolation is an established Thing and definitely what you have used. Keep it :+1: